src/core/gui/XournalView.cpp

```diff
@@ -287,6 +289,20 @@ auto XournalView::onKeyPressEvent(GdkEventKey* event) -> bool {
         return true;
     }
 
+    // Switch color on number key
+    auto &colors = control->getWindow()->getToolMenuHandler()->getColorToolItems();
+    if ((event->keyval >= GDK_KEY_0)
+        && (event->keyval < GDK_KEY_0 + std::min((std::size_t)10,
+                                                 colors.size()))) {
+            std::size_t index = std::min(colors.size() - 1,
+                                         (std::size_t)(9 + (event->keyval - GDK_KEY_0)) % 10);
+            auto colorToolItem = colors.at(index);
```

Comment:
nit: for readability, please explicitly indicate that this is a pointer:
```suggestion
            auto *colorToolItem = colors.at(index);
```